app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt

```diff
@@ -59,6 +75,42 @@ class AppRequestInterceptor(private val context: Context) : RequestInterceptor {
         return RequestInterceptor.ErrorResponse.Uri(errorPageUri)
     }
 
+    /**
+     * Checks if the provided [uri] is a request to install an add-on from addons.mozilla.org and
+     * redirects to Add-ons Manager to trigger installation if needed.
+     *
+     * @return [RequestInterceptor.InterceptionResponse.Deny] when installation was triggered and
+     * the original request can be skipped, otherwise null to continue loading the page.
+     */
+    private fun interceptAmoRequest(
+        uri: String,
+        isSameDomain: Boolean,
+        hasUserGesture: Boolean
+    ): RequestInterceptor.InterceptionResponse? {
+        // First we execute a quick check to see if this is a request we're interested in i.e. a
+        // request triggered by the user and coming from AMO.
+        if (hasUserGesture && isSameDomain && uri.startsWith(AMO_BASE_URL)) {
```

Comment:
This seems potentially insecure. `AMO_BASE_URL` doesn't end with a slash.

Reply:
Could you be a bit more explicit or give an example? :)

Reply:
`AMO_BASE_URL` is `https://addons.mozilla.org`. So when `uri` is for example `https://addons.mozilla.org.evil.com`, `uri.startsWith(AMO_BASE_URL)` would be `true`.

Reply:
Thanks. I will update this in a bit. The actual match against the URL is requiring the `/` though so it shouldn't be insecure as is, but good to change.

Reply:
We are additionally checking against [AMO_INSTALL_URL_REGEX](https://github.com/mozilla-mobile/fenix/pull/16818/files#diff-38274d9212603acf816a7fa82c02e6820dc25a8ed3a489e50b5e57ac9f6ca205R172)  on line [95](https://github.com/mozilla-mobile/fenix/pull/16818/files#diff-38274d9212603acf816a7fa82c02e6820dc25a8ed3a489e50b5e57ac9f6ca205R95)

Reply:
Ah you are right. Sorry

Reply:
No, thanks for taking a look ❤️ !